executor/analyze.go

```diff
@@ -206,9 +213,62 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			}
 		}
 	}
+	err = e.saveAnalyzeOptsV2()
+	if err != nil {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+	}
 	return statsHandle.Update(e.ctx.GetInfoSchema().(infoschema.InfoSchema))
 }
 
+func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
+	if !variable.PersistAnalyzeOptions.Load() || e.OptionsMap == nil {
+		return nil
+	}
+	for _, options := range e.OptionsMap {
+		err := saveAnalyzeOpts(options, e.ctx.(sqlexec.RestrictedSQLExecutor))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func saveAnalyzeOpts(opts core.V2AnalyzeOptions, exec sqlexec.RestrictedSQLExecutor) error {
+	sampleNum := uint64(0)
+	if val, ok := opts.RawOpts[ast.AnalyzeOptNumSamples]; ok {
+		sampleNum = val
+	}
```

Comment:
We can use `sampleNum = opts.RawOpts[ast.AnalyzeOptNumSamples]` directly here since Golang can guarantee the non-existed value of a map is the zero-value of the corresponding type(which is 0 for uint64).

Reply:
Done